src/hotspot/share/gc/shenandoah/shenandoahEvacTracker.cpp

```diff
@@ -53,14 +57,18 @@ void ShenandoahEvacuationStats::accumulate(const ShenandoahEvacuationStats* othe
   _bytes_completed += other->_bytes_completed;
   _evacuations_attempted += other->_evacuations_attempted;
   _bytes_attempted += other->_bytes_attempted;
-
-  _age_table.merge(&other->_age_table);
+  if (_generational && GenShenCensusAtEvac) {
+    // Discard const
+    _age_table.merge(((ShenandoahEvacuationStats*)other)->age_table());
```

Comment:
Is the cast necessary?

Reply:
Avoided the cast by moving age table out of the structure (which makes sense anyway since it's not used in some cases).